apps/ai-image-tagging/frontend/src/components/AITagView/AITagView.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { CheckboxField, TextInput, Pill, Button } from '@contentful/forma-36-react-components';
+import get from 'lodash.get';
+
+import { styles } from './styles';
+
+async function callAPI(url) {
+  const res = await fetch(`/tags/${url}`);
+  const data = await res.json();
+  return data.tags;
+}
+
+export class AITagView extends React.Component {
+
+  constructor(props){
+    super(props);
+
+    this.state = {
+      value: "",
+      checked: false,
+      tags: props.entries.imageTags.getValue() || [],
+      overwrite: true,
+      isMissingImage: !props.entries.image.getValue(),
+      isFetchingTags: false
+    }
+  }
+
+  componentDidMount() {
+    this.props.entries.image.onValueChanged(() => {
+      this.setState(() => ({
+        isMissingImage: !this.props.entries.image.getValue()
+      }))
+    });
+  }
+
+  toggleOverwrite = () => {
+    this.setState(state => ({
+      overwrite: !state.overwrite
+    }));
+  }
+
+  addTag = async (e) => {
+    if (e.key !== "Enter" || !e.target.value) { return; }
+
+    const newTags = [e.target.value, ...this.state.tags];
+    await this.props.entries.imageTags.setValue(newTags);
+    this.setState({
+      tags: newTags,
+      value: ""
+    });
+  }
+
+  deleteTag = async (tag) => {
+    const newTags = this.state.tags.filter(t => t !== tag);
+    await this.props.entries.imageTags.setValue(newTags);
+    this.setState({
+      tags: newTags
+    });
+  }
+
+  fetchTags =  async () => {
+    const imageId = get(this.props.entries.image.getValue(), 'sys.id')
+    const file = await this.props.space.getAsset(imageId);
+    const locale = this.props.locale;
+    const fullURL = get(file, `fields.file.${locale}.url`);
+    const imagePath = new URL(`${location.protocol}${fullURL}`).pathname;
+    this.setState({ isFetchingTags: true });
+
+    try {
+      const aiTags = await callAPI(imagePath);
+
+      // upload new tags
+      const newTags = this.state.overwrite ? aiTags : [...aiTags, ...this.state.tags];
+      await this.props.entries.imageTags.setValue(newTags)
+      this.setState({
+        tags: newTags
+      });
+    } catch(e) {
+      this.props.notifier.error(e.message)
```

Comment:
We should display a crafted message rather than what is included in the `Error` instance. `Tagging image failed.` would be a good start, if you can distinguish between things that can go wrong (here I assume we may fail on HTTP in `callAPI`) then I'd differentiate messages

Reply:
Updated it